### PR TITLE
Add access to boxSearch

### PR DESCRIPTION
**Pull request by gui2dev**

### CODE REVIEW

**Comment by mvieth** [comment]
Some ideas/things I noticed:
- instead of the `std::vector<int>`, the new `Indices` could be used, but since `boxSearch` of `tree_` uses the former as parameter type, a few more changes would be necessary
- it seems like it would make sense if the return type would be unsigned, maybe `unsigned int` or `std::size_t`, but same issue as before
- the indentation seems to be different from the rest of the file

@kunaltyagi I will let you decide what, if anything, you want changed.

**Comment by mvieth** [comment]
> The return value could be index_t as well but it seems a bit weird unless the tree is modified as well (L283 octree_search.h)

I disagree on that one. The returned value isn't an index and should not be used as one. [There is actually a cast from size_t to int.](https://github.com/PointCloudLibrary/pcl/blob/master/octree/include/pcl/octree/impl/octree_search.hpp#L218)

**Comment by SergioRAgostinho** [comment]
> The returned value isn't an index and should not be used as one.

For convenient comparison and arithmetic, your index and size types should be the same. The following idiom is used often
```
for (index_type i = 0; i < size_type ; ++i)
```

**Comment by gui2dev** [comment]
@kunaltyagi I changed the value for `Indices`

File: search/include/pcl/search/octree.h [hunk 1]
```diff
@@ -276,16 +276,16 @@ namespace pcl
...
-		    {
-			    return (tree_->boxSearch(min_pt, max_pt, k_indices));
-		    }
+	  * \param[in] min_pt lower corner of search area
+	  * \param[in] max_pt upper corner of search area
+	  * \param[out] k_indices the resultant point indices
+	  * \return number of points found within search area
+	*/
+	inline int
+	boxSearch(const Eigen::Vector3f &min_pt, const Eigen::Vector3f &max_pt, index_t &k_indices) const
```
Comment (SergioRAgostinho) [inline search/include/pcl/search/octree.h]:
```suggestion
      boxSearch(const Eigen::Vector3f &min_pt, const Eigen::Vector3f &max_pt, Indices &k_indices) const
```

Comment (gui2dev) [reply search/include/pcl/search/octree.h]: 26cc4471270b9d29d8b521a33a438def30aa7a3e

File: search/include/pcl/search/octree.h [hunk 2]
```diff
@@ -275,7 +275,17 @@ namespace pcl
...
           return (tree_->approxNearestSearch (query_index, result_index, sqr_distance));
         }
-
+        /** \brief Search for points within rectangular search area
+	  * \param[in] min_pt lower corner of search area
+	  * \param[in] max_pt upper corner of search area
+	  * \param[out] k_indices the resultant point indices
+	  * \return number of points found within search area
+	*/
+	inline int
```
Comment (SergioRAgostinho) [inline search/include/pcl/search/octree.h:284]: return an `index_t` here

File: search/include/pcl/search/octree.h [hunk 3]
```diff
@@ -275,7 +275,17 @@ namespace pcl
...
+	boxSearch(const Eigen::Vector3f &min_pt, const Eigen::Vector3f &max_pt, Indices &k_indices) const
+	{
+	  return (tree_->boxSearch(min_pt, max_pt, k_indices));
+	}
```
Comment (SergioRAgostinho) [inline search/include/pcl/search/octree.h:288]: The indentation is misaligned from the methods above. You're using tabs instead of spaces.

**Comment by kunaltyagi** [comment]
Pushing to 1.12.0

**Comment by larshg** [comment]
@gui2dev Would you do the required changes?

**Comment by gui2dev** [comment]
@larshg can do.
Just need some refresh

**Comment by gui2dev** [comment]
@larshg, I guess that should do the trick

**Review by kunaltyagi — APPROVED** [review]

**Review by mvieth — APPROVED** [review]

**Pull request merged**